.github/workflows/.swiftlint.yaml

```diff
@@ -0,0 +1,37 @@
+# SwiftLint Configuration File
```

Comment:
Really appreciate you adding all the config options in here and also documenting what will cause errors and warnings once they're turned on, @henrytwagner! Will really help us organize the work for this going forward 😊

Reply:
I'll likely move this file to the project root, but for the purposes of this PR it's totally fine to have it here. I figure this is where ESLint and Prettier configs normally are. Will send a commit with this along later :)